### PR TITLE
Integrate database

### PR DESCRIPTION
**Pull request by oreogunleye24**
The database, frontend, and backend run in a containerized Docker environment.
All backend models, controllers, and services were adjusted to work with the new database structure. 
Adjusted all the frontend APIs
Fixed some design issues.

### CODE REVIEW

File: backend/Models/Family.cs [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
-    public class Family
-    {
-        public Guid uuid { get; set; }
-        public string? name { get; set; }
-        public Guid super_tag_mechanism_uuid { get; set; }
-        public bool isDel {  get; set; }
-    }
+    [Key]
+    [Column("id")]
+    public int Id { get; set; }
```
Comment (K20shores) [inline backend/Models/Family.cs:15]: I have a preference for primary keys to be Guids, as the students did before. Can you make that change here and for all other classes?

File: backend/Models/ReactionSpecies.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    [Column("quantity")]
+    public decimal Quantity { get; set; }
+
+    [ForeignKey("ReactionId")]
+    [InverseProperty("ReactionSpecies")]
+    public virtual Reaction Reaction { get; set; } = null!;
+
+    [ForeignKey("SpeciesId")]
+    [InverseProperty("ReactionSpecies")]
+    public virtual Species Species { get; set; } = null!;
```
Comment (K20shores) [inline backend/Models/ReactionSpecies.cs:36]: `reaction_id`, `species_id` seem to be duplicates of `ReactionId` and `SpeciesId`, why? Also, why do we need this file and `Species.cs`?

**Review by K20shores — CHANGES_REQUESTED** [review]
Just a few comments from my brief review.

File: backend/Models/ReactionSpecies.cs (hunk 2 above)
Comment (oreogunleye24) [reply backend/Models/ReactionSpecies.cs:36]: No, those just say that we have foreign keys in the file. The foreign keys are the reaction_id and the species_id which come from the reaction and species table.

File: backend/Models/Reaction.cs [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
-        public Guid product_list_uuid { get; set; }
-        public string reaction_string {  get; set; }
-    }
+    [Key]
+    [Column("id")]
+    public int Id { get; set; }
+
+    [Column("equation")]
+    [StringLength(512)]
+    public string Equation { get; set; } = null!;
```
[the remark this answers is not in the document]
Comment (oreogunleye24) [reply backend/Models/Reaction.cs:19]: Equation is basically the reaction string.

File: backend/Models/Family.cs (hunk 1 above)
Comment (oreogunleye24) [reply backend/Models/Family.cs:15]: Sure, I'll make that change.

File: backend/Models/Reaction.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (K20shores) [reply backend/Models/Reaction.cs:19]: That's not something I'm expecting, especially if it's something we later parse. What do we gain from storing the equation?

[the remark this answers is not in the document]
Comment (oreogunleye24) [reply backend/Models/Reaction.cs:19]: We need to know what species are involved in the reaction. This information also helps us display the reactions associated with mechanisms and families on the front page.
All the reactions, species, and families I have in the database right now are from this link you gave us: https://github.com/NCAR/music-box/tree/main/src/acom_music_box/examples/configs/ts1

[the remark this answers is not in the document]
Comment (K20shores) [reply backend/Models/Reaction.cs:19]: Right, the species and products of a reaction should not be something that we parse. It should be a relation in the database. I believe that's how the students set it up last semester. I am fine with a more flat database structure with fewer tables, but list relationships still need to be maintained

[the remark this answers is not in the document]
Comment (oreogunleye24) [reply backend/Models/Reaction.cs:19]: Okay that's fine, I will adjust it.

**Pull request merged**